snapcraft/internal/lxd/_cleanbuilder.py

```diff
@@ -50,24 +50,26 @@ def _ensure_container(self):
         self._configure_container()
         self._wait_for_network()
         self._container_run(["apt-get", "update"])
-        # Because of https://bugs.launchpad.net/snappy/+bug/1628289
-        # Needed to run snapcraft as a snap and build-snaps
-        self._container_run(["apt-get", "install", "squashfuse", "-y"])
-        self._inject_snapcraft(new_container=True)
+        self._inject_snapcraft()
 
     def _setup_project(self):
-        logger.info("Setting up container with project assets")
         tar_filename = self._source
         # os.sep needs to be `/` and on Windows it will be set to `\`
         dst = "{}/{}".format(self._project_folder, os.path.basename(tar_filename))
         self._container_run(["mkdir", self._project_folder])
-        self._push_file(tar_filename, dst)
+        self._lxd_instance.push_file(tar_filename, dst)
         self._container_run(
             ["tar", "xvf", os.path.basename(tar_filename)], cwd=self._project_folder
         )
 
-    def _finish(self):
-        # os.sep needs to be `/` and on Windows it will be set to `\`
-        src = "{}/{}".format(self._project_folder, self._snap_output)
-        self._pull_file(src, self._snap_output)
-        logger.info("Retrieved {}".format(self._snap_output))
+    def _finish(self, success=True):
+        super()._finish(success=success)
+
+        # remove project data as this is a cleanbuild
+        if os.path.exists(self.provider_project_dir):
+            shutil.rmtree(self.provider_project_dir)
```

Comment:
I like this, but I suggest we also put the tarball we generated of the project in there so we can toast it when we're done. If that doesn't work for some reason, can we at least also remove the tarball here?

Reply:
I am going to say no to this one, this is not where the tarball goes either.
This is the directory of the provider for private project things like the snap-registry.yaml (which if left would cause bad results) and such. The tarball could have usefulness on good or bad outcomes and is part of the original experience.